### PR TITLE
Add `message` attribute to run api

### PR DESCRIPTION
**Pull request by caseylang**
`message` is now a valid Run API attribute and will be persisted when passed in. It is also optional. The run will have either if the action will destroy `Queued manually to destroy infrastructure`, otherwise we store `Queued manually via the Terraform Enterprise API`.
Relates to https://github.com/hashicorp/atlas/pull/6072

### CODE REVIEW

File: content/source/docs/enterprise/api/run.html.md [hunk 1]
```diff
@@ -39,6 +40,7 @@ A run performs a plan and apply, using a configuration version and the workspace
   "data": {
     "attributes": {
       "is-destroy":false
+      "message": "Custom message"
```
Comment (nfagerlund) [inline content/source/docs/enterprise/api/run.html.md]: json syntax error, needs a comma to separate it from the previous property!

File: content/source/docs/enterprise/api/run.html.md [hunk 2]
```diff
@@ -29,6 +29,7 @@ A run performs a plan and apply, using a configuration version and the workspace
 
 - `id` (`string: <required>`) - specifies the workspace ID to run
 - `is-destroy` (`bool: false`) - specifies if this plan is a destroy plan, which will destroy all provisioned resources.
+- `message` (`string: <optional`) - specifies the message to be associated with this run.
```
Comment (nfagerlund) [inline content/source/docs/enterprise/api/run.html.md]: Hey, while you're in this file... could I get you to reformat this method to follow `_template.md`? 😄

**Comment by caseylang** [comment]
All set @nfagerlund

**Comment by nfagerlund** [comment]
@caseylang Sweet, thank you! I pushed one fixup commit — there were some markdown syntax errors, and the type is "runs" instead of "run."

File: content/source/docs/enterprise/api/run.html.md [hunk 3]
```diff
@@ -36,11 +36,17 @@ Key path                    | Type   | Default | Description
...
+Status  | Response                               | Reason
+--------|----------------------------------------|-------
+[200][] | [JSON API document][] (`type: "runs"`) | Successfully created a run
+[404][] | [JSON API error object][]              | Organization or workspace not found, or user unauthorized to perform action
+[422][] | [JSON API error object][]              | Validation errors
+
+[JSON API document]: https://www.terraform.io/docs/enterprise/api/index.html#json-api-documents
+[200]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/200
+[404]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/404
+[422]: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422
```
Comment (caseylang) [inline content/source/docs/enterprise/api/run.html.md]: ah ha! I was wondering what was going on in the template. I've never seen this link style before. Good to know

**Pull request merged**